consensus/src/consensus_observer/observer/consensus_observer.rs

```diff
@@ -779,26 +796,15 @@ impl ConsensusObserver {
                 metrics::ORDERED_BLOCK_LABEL,
             );
 
-            if self.pipeline_enabled() {
-                for block in ordered_block.blocks() {
-                    let commit_callback = self.active_observer_state.create_commit_callback(
-                        self.ordered_block_store.clone(),
-                        self.block_payload_store.clone(),
-                    );
-                    self.pipeline_builder().build(
-                        block,
-                        self.get_last_pipeline_futs(),
-                        commit_callback,
-                    );
-                }
-            }
             // Insert the ordered block into the pending blocks
             self.ordered_block_store
                 .lock()
                 .insert_ordered_block(ordered_block.clone());
 
             // If state sync is not syncing to a commit, finalize the ordered blocks
-            if !self.state_sync_manager.is_syncing_to_commit() {
+            if !self.state_sync_manager.is_syncing_to_commit()
+                && !self.state_sync_manager.in_fallback_mode()
```

Comment:
Nit: this is probably not needed. When we enter fallback mode, we terminate all subscriptions, so we never process any CO network messages in fallback mode.

Reply:
hmm i added this because i saw in logs that both fallback and pipeline are running

Reply:
Weird. Do you have a log trace? When we fallback, we terminate all subscriptions, so any messages received after that should be dropped. Wondering if there's a race condition or something else 🤔